chromium_src/ui/views/controls/focus_ring.cc

```diff
@@ -52,11 +45,59 @@ FocusRingTheme& GetFocusRingTheme() {
 
 }  // namespace
 
-#define setStyle                                                           \
-  setColor(color_.value_or(                                                \
-      GetFocusRingTheme().GetSystemColor(ColorIdForValidity(!invalid_)))); \
-  paint.setStyle
+#define Install Install_UnUsed
 
 #include "../../../../../ui/views/controls/focus_ring.cc"
 
-#undef setStyle
+#undef Install
+
+namespace views {
+
+class BraveFocusRing : public FocusRing {
+ public:
+  using FocusRing::FocusRing;
+
+  BraveFocusRing(const BraveFocusRing&) = delete;
+  BraveFocusRing& operator=(const BraveFocusRing&) = delete;
+
+  // Copied all of FocusRing::OnPaint and changed setColor only.
+  // FocusRing overrides:
+  void OnPaint(gfx::Canvas* canvas) override {
+    // TODO(pbos): Reevaluate if this can turn into a DCHECK, e.g. we should
+    // never paint if there's no parent focus.
+    if (has_focus_predicate_) {
+      if (!(*has_focus_predicate_)(parent()))
+        return;
+    } else if (!parent()->HasFocus()) {
+      return;
+    }
+
+    const SkRRect ring_rect = GetRingRoundRect();
+    cc::PaintFlags paint;
+    paint.setAntiAlias(true);
+    paint.setStyle(cc::PaintFlags::kStroke_Style);
+
+    if (parent()->GetProperty(kDrawFocusRingBackgroundOutline)) {
+      // Draw with full stroke width + 2x outline thickness to effectively paint
+      // the outline thickness on both sides of the FocusRing.
+      paint.setStrokeWidth(FocusRing::kHaloThickness + 2 * kOutlineThickness);
+      paint.setColor(GetCascadingBackgroundColor(this));
+      canvas->sk_canvas()->drawRRect(ring_rect, paint);
+    }
+
+    paint.setColor(color_.value_or(
+        GetFocusRingTheme().GetSystemColor(ColorIdForValidity(!invalid_))));
```

Comment:
if this is the only thing that changed, then you can create `FocusRing::GetColor` and return the desired color instead.
copy/pasting upstream logic can lead to other bugs.

Reply:
you can also try to subclass PaintFlags here and do something with it.

Reply:
Wow, great idea 👍🏼 
At first, I thought redefine `GetColor()` in anonymous namespace but it was hard.